proxy/envoy/watcher.go

```diff
@@ -194,18 +193,24 @@ func envoyArgs(fname string, epoch int, mesh *proxyconfig.ProxyMeshConfig, node
 	}
 }
 
-func runEnvoy(mesh *proxyconfig.ProxyMeshConfig, node, configpath string) proxy.Proxy {
+func runEnvoy(mesh *proxyconfig.ProxyMeshConfig, node, configpath, customConfig string) proxy.Proxy {
 	return proxy.Proxy{
 		Run: func(config interface{}, epoch int, abort <-chan error) error {
 			envoyConfig, ok := config.(*Config)
 			if !ok {
 				return fmt.Errorf("Unexpected config type: %#v", config)
 			}
 
-			// attempt to write file
-			fname := configFile(configpath, epoch)
-			if err := envoyConfig.WriteFile(fname); err != nil {
-				return err
+			var fname string
+			if len(customConfig) > 0 {
```

Comment:
I had to stop and think for awhile whether this has implication on the certificate revocation workflow. The agent inserts the SHA of certs into the generated config. With a custom config, the SHA cannot be inserted. Seems like it's still not a problem since you trap it just before the binary call. Please document this since it's non-trivial.